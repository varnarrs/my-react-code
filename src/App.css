
  .App{
    text-align: center;
  }

  .App-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
  } 

  .App-header {
    background-color: #222;
    height: 150px;
    padding: 20px;
    color: white;
  }

  .App-title {
    font-size: 1.5em;
  }

  .App-intro {
    font-size: large;
  }

  @keyframes App-logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  }

  .listStyleType{
    list-style-type: none;
    padding: 0;
  }

  .radioContainer{
    padding-bottom: 10px;
  }

  .showDropdown{
    display: block;
    width:30%;
    border: 1px solid lightgray;
    margin:0 20px  20px 20px;
    padding: 0 10px;
  }

  .hideDropdown{
    display: none;
  }

  input[type=checkbox]:not(old),
  input[type=radio   ]:not(old){
    width     : 2em;
    margin    : 0;
    padding   : 0;
    font-size : 1em;
    opacity   : 0;
  }

  input[type=checkbox]:not(old) + label,
  input[type=radio   ]:not(old) + label{
    display      : inline-block;
    margin-left  : -2em;
    line-height  : 1.5em;
  }

  input[type=checkbox]:not(old) + label > span,
  input[type=radio   ]:not(old) + label > span{
    display          : inline-block;
    width            : 0.875em;
    height           : 0.875em;
    margin           : 0.25em 0.5em 0.25em 0.25em;
    border           : 0.0625em solid lightgrey;
    border-radius    : 0.25em;
    background       : lightgrey;   
    vertical-align   : bottom;
  }

  input[type=radio   ]:not(old) + label > span{
    border-radius    : 100%;
  }
 
  input[type=checkbox]:not(old):checked + label > span:before{
    content     : 'âœ“';
    display     : block;
    width       : 1em;
    color       : blue;
    font-size   : 0.875em;
    line-height : 1em;
    text-align  : center;
    text-shadow : 0 0 0.0714em lightgrey;
    font-weight : bold;
  }

  input[type=radio]:not(old):checked + label > span > span{
    display          : block;
    width            : 0.5em;
    height           : 0.5em;
    margin           : 0.125em;
    border           : 0.0625em solid blue;
    border-radius    : 100%;
    background       : blue;
  
  }

  .multiSelectContainer{
    text-align: left;
  }

  .inputContainer{
    width: 30%;
    padding: 10px;
    margin: 20px 20px 0px 20px;
  }

  .inputContainer:before{
    content     : '&#xf107';
    display     : block;
    width       : 1em;
    color       : blue;
    font-size   : 0.875em;
    line-height : 1em;
    text-align  : center;
    text-shadow : 0 0 0.0714em lightgrey;
    font-weight : bold;
  }

  .buttonContainer{
    padding: 10px;
    background-color: lightgrey;
    margin: -10px -10px -15px -10px;
  }

  .styledSelect {
    font-weight: bold;
  }

.styledSelect:after {
  content: "";
  width: 0;
  height: 0;
  border: 7px solid transparent;
  border-color: blue transparent transparent transparent;
  position: relative;
  top: 12px;
  right: 40px;
}

.liContainer{
  height: auto;
    max-height: 300px;
    overflow-y: auto;
}

.radioPosition{
  padding-left: 40px;
}