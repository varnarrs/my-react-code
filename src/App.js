import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

import MultiSelectBox from './Multiselect.js'
// import FabricDropdown from './FabricDropdown.js'


class App extends Component {

  constructor() {
    super();
    this.state = {

      selectBoxData: []
    };
  }



  componentDidMount = () => {
    let selectBoxData = [
      {
        key: 'corn',
        value: 'Corn',
        text: 'Corn'
      },
      {
        key: 'capsicum',
        value: 'Capsicum',
        text: 'Capsicum'
      },
      {
        key: 'onion',
        value: 'Onions',
        text: 'Onions'
      },
      {
        key: 'paneer',
        value: 'Paneer',
        text: 'Paneer'
      }
    ];
    this.setState({ selectBoxData: selectBoxData });
  }

  onSelectedItems = (items) => {
    console.log(items);
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <MultiSelectBox
          selectedItems={this.onSelectedItems}
          selectBoxData={this.state.selectBoxData} />
        {/* <div>
          <FabricDropdown selectBoxData={this.state.selectBoxData} />
        </div> */}
      </div>
    );
  }
}

export default App;
