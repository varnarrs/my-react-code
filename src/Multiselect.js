

import './App.css';
import React, { Component } from 'react';

class MultiSelectBox extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            showDropdown: false,
            inputValue: ''
        };
        this.selectedItems = [];
        this.checked = {};
        this.radioChecked = 'or';
    }

    onCheckBoxClick = (e) => {
        let selVal = e.target.value;
        if (e.target.checked) {
            this.checked[selVal] = true;
            this.selectedItems.push(selVal);
        } else {
            this.checked[selVal] = false;
            const currIndex = this.selectedItems.indexOf(selVal);
            if (currIndex > -1) {
                this.selectedItems.splice(currIndex, 1);
            }
        }
        this.setInputValue();

    }
    setInputValue = () => {
        let selectedItemsVal = '';
        this.selectedItems.map((val) => {
            if (!selectedItemsVal) {
                selectedItemsVal = val;
            } else {
                selectedItemsVal = selectedItemsVal + ' ' + this.radioChecked + ' ' + val;
            }
        });
        this.setState({ inputValue: selectedItemsVal });
        this.props.selectedItems(selectedItemsVal);
    }
    toggleDropdown = () => {
        this.setState({ showDropdown: !this.state.showDropdown });
    }
    closeDropdown = () => {
        this.setState({ showDropdown: false });
    }
    onradioClick = (e) => {
        this.radioChecked = e.target.id;
        this.setInputValue();

    }
    clearSelection = () => {
        this.selectedItems = [];
        this.setState({ inputValue: '' });
        for (let check in this.checked) {
            this.checked[check] = false;
        }
    }


    render() {
        return (
            <div className={'multiSelectContainer'}>
                <div className={'styledSelect'}>
                    <input type="text" className={'inputContainer'} onClick={this.toggleDropdown}
                        placeholder={'Select Toppings'} defaultValue={this.state.inputValue}
                        title={this.state.inputValue} />
                </div>
                <div className={this.state.showDropdown ? 'showDropdown' : 'hideDropdown'}>
                    <ul className={'listStyleType'} >
                        <li className='radioContainer'>
                            <input type="radio"
                                name="selectType"
                                id="or"
                                value="Any"
                                onClick={this.onradioClick}
                                defaultChecked />
                            <label htmlFor="any"><span><span></span></span>{'Any'}</label>
                            <input type="radio"
                                name="selectType"
                                id="and"
                                onClick={this.onradioClick}
                                value="All" />
                            <label htmlFor="all"><span><span></span></span>{'All'}</label>
                        </li>
                        <div className={'liContainer'}>
                            {this.props.selectBoxData.map((val) => {
                                return (<li value={val.value}>
                                    <input type="checkbox"
                                        defaultValue={val.value}
                                        id={'check_' + val.value}
                                        checked={this.checked[val.value]}
                                        onClick={this.onCheckBoxClick} />
                                    <label htmlFor={'check_' + val.value}><span><span></span></span>{val.value}</label>
                                </li>);
                            })}
                        </div>
                        <li>

                            <hr />
                            <div className={'buttonContainer'} onClick={this.clearSelection}> {'X Clear'}</div>


                        </li>
                    </ul>
                </div>
            </div>
        );
    }
}
export default MultiSelectBox;