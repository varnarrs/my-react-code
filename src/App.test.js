import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Multiselect from './Multiselect';
import { shallow, configure } from 'enzyme';  
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App | initial loading and methods with shallow', () => {
  let output;
  beforeEach(() => {
    output = shallow(<App />);
  });

  it('App | Should match snapshot to avoid sudden/unnecessary changes', () => {
    //Checking with snapshot (created on first run) to avoid any unnecessary/unintended sudden changes to UI
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('App | is Testing initial loading and it should have Multiselectbox ', () => {
    output = shallow(<App />);
    expect(output.find(Multiselect)).toHaveLength(1);
  });

  it('App | componentDidMount() ', () => {
    output = shallow(<App />);
    output.instance().componentDidMount();
  });
});
