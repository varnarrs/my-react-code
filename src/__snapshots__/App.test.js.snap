// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App | initial loading and methods with shallow App | Should match snapshot to avoid sudden/unnecessary changes 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <MultiSelectBox
    selectBoxData={
      Array [
        Object {
          "key": "corn",
          "text": "Corn",
          "value": "Corn",
        },
        Object {
          "key": "capsicum",
          "text": "Capsicum",
          "value": "Capsicum",
        },
        Object {
          "key": "onion",
          "text": "Onions",
          "value": "Onions",
        },
        Object {
          "key": "paneer",
          "text": "Paneer",
          "value": "Paneer",
        },
      ]
    }
    selectedItems={[Function]}
  />
</div>
`;
